app/bluetrax.py

```
@@ -117,10 +117,16 @@ async def get_asset_history(unit_id: str, start_time: datetime, end_time: dateti
           }
     
     async with httpx.AsyncClient() as client:
-        r = await client.get("https://rest.bluetrax.co.ke/AnalyticsService", 
+        r = await client.post("https://rest.bluetrax.co.ke/AnalyticsService", 
                              params={'request': json.dumps(q)}
         )
-        if r.status_code == 200:
-            return HistoryResult(**r.json())
-        else:
-            return None
+
+        if r.is_success:
+            try:
+                data = r.json()
+                history = HistoryResult.validate(data)
+                return history
+            except pydantic.ValidationError as ve:
+                raise ve
+
+        r.raise_for_status()
```

This logic to swallow errors prevent me from seeing the change in Bluetrax's API.
It now let's a non-200 response raise an exception to be handled by activity logging.

---

I'm just wondering, if we catch `ValidationError` only to raise it again, why do we catch it? Maybe we could log something about the error here (e.g. the response data if that's useful for detecting schema changes in the API response).

---

Yes, that makes sense. I was focused on HTTP errors that I didn't reconsider the handling for validation failure. I'll follow up with a change for this.